### PR TITLE
Update documents for release process

### PR DESCRIPTION
**Pull request by fgrzadkowski**
Some minor changes to the release process documents which were not obvious after I did the release.

@quinton-hoole

### CODE REVIEW

File: docs/devel/releasing.md [hunk 1]
```diff
@@ -208,9 +208,9 @@ release](https://github.com/kubernetes/kubernetes/releases/new):
 1. fill in the release title from the draft;
 1. re-run the appropriate release notes tool(s) to pick up any changes people
    have made;
-1. find the appropriate `kubernetes.tar.gz` in GCS, download it, double check
-   the hash (compare to what you had in the release notes draft), and attach it
-   to the release; and
+1. find the appropriate `kubernetes.tar.gz` in [GCS bucket](https://pantheon.corp.google.com/storage/browser/kubernetes-release/release/),
```
Comment (ihmccreery) [inline docs/devel/releasing.md]: Do we care that this link is public?

**Comment by ihmccreery** [comment]
LGTM once comment is addressed; @fgrzadkowski please feel free to apply LGTM label once you've addressed that comment.

**Comment by ihmccreery** [comment]
Thanks for the fixups!

**Comment by ihmccreery** [comment]
Ping @fgrzadkowski.

**Comment by ihmccreery** [comment]
Can you also add a conditional for alpha releases so that `RELEASE_BRANCH` gets set to `master`, so the docs are correctly pointed at HEAD of `master`, per @bgrant0607?

**Comment by ihmccreery** [comment]
Ping @fgrzadkowski!

File: docs/devel/releasing.md (hunk 1 above)
Comment (fgrzadkowski) [inline docs/devel/releasing.md]: I don't understand. Its's public anyway.

**Comment by ihmccreery** [comment]
@k8s-bot unit test this please

**Pull request merged**